stacks/upload-api-stack.js

```diff
@@ -55,9 +55,9 @@ export function UploadApiStack({ stack, app }) {
   const git = getGitInfo()
   const ucanInvocationPostbasicAuth = new Config.Secret(stack, 'UCAN_INVOCATION_POST_BASIC_AUTH')
 
-  const apis = (customDomains ?? [undefined]).map((customDomain) => {
+  const apis = (customDomains ?? [undefined]).map((customDomain, idx) => {
     const hostedZone = customDomain?.hostedZone
-    const apiId = [`http-gateway`, hostedZone?.replaceAll('.', '_')]
+    const apiId = [`http-gateway`, idx > 0 ? hostedZone?.replaceAll('.', '_') : '']
```

Comment:
is the idea here that the first customDomain will be web3.storage (which we don't want to change the apiId for per the PR description) - seems fine to me but could we add this comment?

```suggestion
    // the first customDomain will be web3.storage, and we don't want the apiId for that domain to have a second part, see PR of this change for context
    const apiId = [`http-gateway`, idx > 0 ? hostedZone?.replaceAll('.', '_') : '']
```